test/integration/http_integration.cc

```diff
@@ -513,7 +513,8 @@ void HttpIntegrationTest::testRouterUpstreamDisconnectBeforeRequestComplete() {
 
   EXPECT_TRUE(response->complete());
   EXPECT_STREQ("503", response->headers().Status()->value().c_str());
-  EXPECT_EQ("upstream connect error or disconnect/reset before headers", response->body());
+  EXPECT_THAT(response->body(),
+              HasSubstr("upstream connect error or disconnect/reset before headers"));
```

Comment:
Can you actually expect the string with the reset reason? Do we have coverage of the other possible responses somewhere (mostly just want to make sure that all of those case statements are hit and we don't wind up in NOT_REACHED somehow).

Reply:
I have changed this to match with reset reason. Added a test in `utility_test` to verify all cases are covered. But if you want to verify other possible responses via integration test (or some other unit test), I think there are not there. We could add them probably in follow-up PR if needed?